pkg/provider/util/instancetypes/azure.go

```diff
@@ -185,10 +185,13 @@ func filterCPUsAndMemory(cpus, memory int32, arch arch, nestedVirt bool) filterF
 			if nestedVirt && !vm.nestedVirtSupported() {
 				return
 			}
-
 			if vm.VCPUs >= cpus && vm.Memory >= memory && vm.Arch == arch.String() &&
 				vm.baseFeaturesSupported() {
-				vmCh <- vm.Name
+				var lowerCpuPattern = `Standard.*-.*_v\d$`
```

Comment:
Can you add some comment on the expression to tell what is about i.e. 

```
// standard D,E and F series are the VM families
// supporting nested virtualization
var dSeriesPattern = `standardD.*v[3-6]Family$`
```

Probably it also make sense to move it from here to outside of the function, just as a var at the beginning of the file with clear information about what is this used for.

Also consider to extend the commit message to something like:

```
fix: azure wrong capacity for vm sizes. 

Here explain the problem and add the link from the issue (azure link about half capacity machines...)
```

Beyond this LGTM